File: app/src/main/java/com/absinthe/kage/device/CommandBuilder.java
package com.absinthe.kage.device;

import androidx.annotation.NonNull;

public class CommandBuilder {

    private StringBuilder mStringBuilder;

    public CommandBuilder() {
        mStringBuilder = new StringBuilder();
    }

    public CommandBuilder with(Command command) {
        mStringBuilder.append(command.cmd);
        return this;
    }

    public CommandBuilder append(@NonNull String param) {
        mStringBuilder.append(Command.DELIMITER).append(param);
        return this;
    }

    public String build() {
        return mStringBuilder.toString();
    }
}

File: app/src/main/java/com/absinthe/kage/device/CommandBuilder.kt
package com.absinthe.kage.device

class CommandBuilder {

    private val mStringBuilder: StringBuilder = StringBuilder()

    fun with(command: Command): CommandBuilder {
        mStringBuilder.append(command.cmd)
        return this
    }

    fun append(param: String?): CommandBuilder {
        mStringBuilder.append(Command.DELIMITER).append(param)
        return this
    }

    fun build(): String {
        return mStringBuilder.toString()
    }

}